lib/google_search_data_viewer/accounts/user.ex

```diff
@@ -0,0 +1,47 @@
+defmodule GoogleSearchDataViewer.Accounts.User do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "users" do
+    field :email, :string
+    field :password, :string, virtual: true
+    field :hashed_password, :string
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(user, attrs) do
+    user
+    |> cast(attrs, [:email, :password])
+    |> validate_required([:email, :password])
+  end
+
+  @spec register_changeset(
+          {map, map}
+          | %{
+              :__struct__ => atom | %{:__changeset__ => map, optional(any) => any},
+              optional(atom) => any
+            },
+          :invalid | %{optional(:__struct__) => none, optional(atom | binary) => any}
+        ) :: Ecto.Changeset.t()
+  @doc false
```

Comment:
For the IC, we don't require the type spec, however, if you can do it, it's also a good thing to learn.

Reference: https://hexdocs.pm/elixir/1.13/typespecs.html

```suggestion
  @spec register_changeset(%__MODULE__{}, map()) :: Ecto.Changeset.t()
```

If you use `typespec` here, please make sure all other public methods in the `context` also have `typespec`.

Reply:
Notes, I will remove